quarkus/runtime/src/main/java/org/keycloak/provider/quarkus/QuarkusPlatform.java

```diff
@@ -108,4 +114,29 @@ private void addDeferredException(Throwable t) {
         return deferredExceptions;
     }
 
+    @Override
+    public File getTmpDirectory() {
+        if (tmpDir == null) {
+            String homeDir = Environment.getHomeDir();
+
+            File tmpDir;
+            if (homeDir == null) {
+                // Should happen just in the unit tests
+                homeDir = System.getProperty("java.io.tmpdir");
+                tmpDir = new File(homeDir, "keycloak-quarkus-tmp");
+                tmpDir.mkdir();
```

Comment:
This doesn't guarantee that this directory wasn't created by a malicious user.

This logic doesn't guarantee exclusivity. `tmpDir.mkdir()` will return `false` if the directory already exists.

Reply:
Thanks for spotting this. But this "branch" of the source code is triggered just in the rare case when quarkus is executed in the unit tests. As pointed in the above comment. It is the case when you trigger:
```
cd quarkus
mvn clean install
```
in the keycloak sources.

For the case of real quarkus distribution and for the case of integration tests, there is always `Environment.getHomeDir()` return not-null. You can check with the quarkus distribution

Reply:
If this code is executed in the context of a unit test, it's still a vulnerability. It's just a vulnerability in the unit test, right?

IE. If users are writing tests that navigate this code, shouldn't they also be protected?

As an example, this is a vulnerability I found in JUnit 4.
https://github.com/junit-team/junit4/security/advisories/GHSA-269g-pwp5-87pp

Reply:
So far, this is rather theoretic vulnerability IMO. The directory used in the unit tests does not contain any sensitive data or anything, which could somehow break the system. However I agree that will be better to make sure that directory is always created exclusively by the user triggering the test.

I've created JIRA and sent the PR https://github.com/keycloak/keycloak/pull/7916. Do you have a chance to take a look please?